demo/midgard-sdk/src/tx-builder/user-events/deposit/index.ts

```diff
@@ -1 +1,65 @@
+import { Data, Integer, LucidEvolution, Script, TxBuilder} from "@lucid-evolution/lucid";
+import { DepositDatum, DepositInfo, MintRedeemer } from "./types.js";
+import { POSIXTime } from "@/tx-builder/common.js";
+import { sha3_256 } from "@noble/hashes/sha3";
+import { Int } from "effect/Schema";
+
+
+export type DepositParams = {
+  depositScriptAddress: string,
+  mintingPolicy: Script,
+  policyId: string,
+  depositInfo: DepositInfo,
+  inclusionTime: POSIXTime,
+};
+
+/**
+ * Deposit
+ *
+ * @param lucid - The LucidEvolution
+ * @param params - The parameters
+ * @returns {TxBuilder} A TxBuilder instance that can be used to build the transaction.
+ */
+
+export const depositTxBuilder = async (
+  lucid: LucidEvolution,
+  params: DepositParams,
+): Promise<TxBuilder> => {
+
+  const redeemer: MintRedeemer = {
+    AuthenticateEvent: {
+        nonce_input_index: 0n,
+        event_output_index: 0n,
+        hub_ref_input_index: 0n,
+        witness_registration_redeemer_index: 0n
+    }
+  };
+  const authenticateEvent = Data.to(redeemer,MintRedeemer);
+  const utxos = await lucid.wallet().getUtxos();
+  if (utxos.length === 0) {
+    throw new Error("No UTxOs found in wallet");
+  }
+  const inputUtxo = utxos[0];
+  const assetName = sha3_256(inputUtxo.txHash);
```

Comment:
After switching to `effect` from `Promise`, this should look like below:
```suggestion
  const assetNameBuffer = yield* hashHexWithBlake2b256(inputUtxo.txHash);
  const assetName = bufferToHex(assetNameBuffer);
```

`bufferToHex` is a helper I've defined in `midgard-node` ([here](https://github.com/Anastasia-Labs/midgard/blob/f37f06bb3d48340e9852ac128d33e800d3e09f18/demo/midgard-node/src/utils.ts#L55-L61)), so feel free to bring it over into `midgard-sdk/utils/common.ts`. Just note that this'll require anywhere this helper has been used in `midgard-node` to change so that it's imported from `midgard-sdk`.